pyccel/parser/scope.py

```diff
@@ -488,6 +499,23 @@ def insert_symbol(self, symbol):
                 self._used_symbols[symbol] = collisionless_symbol
                 self._original_symbol[collisionless_symbol] = symbol
 
+    def remove_symbol(self, symbol):
+        """
+        Remove symbol from the scope.
+
+        Remove symbol from the scope.
+
+        Parameters
+        ----------
+        symbol : PyccelSymbol
+            The symbol to be added to the scope.
+        """
```

Comment:
in the case of an Interface, every time we pass through this [line](https://github.com/pyccel/pyccel/blob/d9bf2a422a9d87a03759cc6504d55d84f85cebaf/pyccel/parser/semantic.py#L3912) a new incremented symbol is added even if the function is not annotated, at the end we end up with function numbers that is almost random because we pass through that line many times depending on when the user calls the function. as opposed to passing through that line one time as in the past.

Reply:
fixed docstring in [b30d725](https://github.com/pyccel/pyccel/pull/1728/commits/b30d725e4cdff8de09df23ceae39eba50465d689)